src/test/java/io/mosip/mimoto/util/CredentialIssuerWellknownResponseValidatorTest.java

```diff
@@ -0,0 +1,233 @@
+        response.getCredentialConfigurationsSupported().get("CredentialType1").setScope(null);
+        response.getCredentialConfigurationsSupported().get("CredentialType1").setDisplay(null);
+        response.getCredentialConfigurationsSupported().get("CredentialType1").setProofTypesSupported(null);
+        response.getCredentialConfigurationsSupported().get("CredentialType1").setCredentialDefinition(null);
+        CredentialIssuerWellknownResponseValidator credentialIssuerWellknownResponseValidator = new CredentialIssuerWellknownResponseValidator();
+        InvalidWellknownResponseException invalidWellknownResponseException = assertThrows(InvalidWellknownResponseException.class, () ->
+                credentialIssuerWellknownResponseValidator.validate(response, validator));
+        Assertions.assertEquals("RESIDENT-APP-041", invalidWellknownResponseException.getErrorCode());
+        Assertions.assertTrue(invalidWellknownResponseException.getMessage().contains("format: Format must not be blank"));
+        Assertions.assertTrue(invalidWellknownResponseException.getMessage().contains("scope: Scope must not be blank"));
+        Assertions.assertTrue(invalidWellknownResponseException.getMessage().contains("proofTypesSupported: Proof types supported must not be empty"));
+        Assertions.assertTrue(invalidWellknownResponseException.getMessage().contains("credentialDefinition: must not be null"));
+        Assertions.assertTrue(invalidWellknownResponseException.getMessage().contains("display: Display information must not be empty"));
+
+    }
+
+    @Test
+    public void shouldDetectMissingProofAlgorithmsSupported() {
+        CredentialIssuerWellKnownResponse response = getCredentialIssuerWellKnownResponseDto("Issuer1",
+                Map.of("CredentialType1", getCredentialSupportedResponse("CredentialType1")));
+        response.getCredentialConfigurationsSupported().get("CredentialType1").getProofTypesSupported().get("jwt");
+        CredentialIssuerWellknownResponseValidator credentialIssuerWellknownResponseValidator = new CredentialIssuerWellknownResponseValidator();
+        InvalidWellknownResponseException invalidWellknownResponseException = assertThrows(InvalidWellknownResponseException.class, () ->
+                credentialIssuerWellknownResponseValidator.validate(response, validator));
+        Assertions.assertTrue(invalidWellknownResponseException.getMessage().contains("proofSigningAlgValuesSupported: must not be null"));
+    }
+
+    @Test
+    public void shouldDetectMissingMandatoryFieldsOfCredentialSupportedDisplayResponse() {
+        // Create a new response object with nested display response details
+        CredentialIssuerWellKnownResponse response = getCredentialIssuerWellKnownResponseDto("Issuer1",
+                Map.of("CredentialType1", getCredentialSupportedResponse("CredentialType1")));
+        List<CredentialSupportedDisplayResponse> credentialSupportedDisplayResponseList = new ArrayList<>();
```

Comment:
Lets Extract the object creation to TestUtilities.

Reply:
I used the base objects that were presnt in test utilities and modified the key value accordingly, to handle different scenarios